src/api/app/models/group.rb

```diff
@@ -34,7 +32,9 @@ class Group < ApplicationRecord
   alias_attribute :name, :title
 
   def self.find_by_title!(title)
-    find_by_title(title) || (raise NotFoundError, "Couldn't find Group '#{title}'")
+    find_by!(title: title)
+  rescue ActiveRecord::RecordNotFound => e
```

Comment:
any better way to change the message of the raised exception?

Reply:
I don't think so.

I've found this [StackOverflow thread](https://stackoverflow.com/questions/2823748/how-do-i-add-information-to-an-exception-message-in-ruby) which is providing more info.

Reply:
I don't think so.

I've found this [StackOverflow thread](https://stackoverflow.com/questions/2823748/how-do-i-add-information-to-an-exception-message-in-ruby) which is providing more info.

Reply:
I take it you checked how the API is handling `ActiveRecord::RecordNotFound` right?

Reply:
@hennevogel 

```
rescue_from Backend::NotFoundError, ActiveRecord::RecordNotFound do |exception|
      render_error message: exception.message, status: 404, errorcode: 'not_found'
    end
```
Is there something missing? Any test that you want to see?

Reply:
Nope, all is good. I was just not sure.

Reply:
```
(base) vpereira@linux-qouy:~/open-build-service> dosc api '/group/xxxx'
WARNING: SSL certificate checks disabled. Connection is insecure!

Server returned an error: HTTP Error 404: Not Found
Couldn't find Group 'xxxx'
```